app/operations/workflows/unretire_subjects.rb

```diff
@@ -17,7 +21,14 @@ def execute
     end
 
     def cached_subject_ids
-      @cached_subject_ids ||= Array.wrap(@subject_ids) | Array.wrap(@subject_id)
+      cached_subject_ids ||= Array.wrap(@subject_ids) | Array.wrap(@subject_id)
+      cached_subject_ids.push(*subject_set_subject_ids) unless subject_set_subject_ids.empty?
+      @cached_subject_ids = cached_subject_ids
```

Comment:
I think we can simplify this a little bit - keep the `cached_subject_ids` method but add a layer underneath it to handle the subject_id lookups, e.g. something like
```
  def cached_subject_ids
    # note the union | array operator below will allow us to accept both sets of params :)
    @cached_subject_ids ||= subject_ids | subject_set_subject_ids
  end

  def subject_ids
      Array.wrap(@subject_ids) | Array.wrap(@subject_id)
  end
```